lib/internal/promisify.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+module.exports = promisify;
+
+function promisify(fn, single) {
+  Object.defineProperty(inner, 'name', {
+    value: fn.name + 'Promisified'
+  });
+  return inner;
+
+  function inner() {
+    if (typeof arguments[arguments.length - 1] === 'function') {
+      return fn.apply(this, arguments);
+    }
+    var resolve = null;
+    var reject = null;
+    const args = Array.from(arguments);
+    const promise = new Promise((_resolve, _reject) => {
+      resolve = _resolve;
+      reject = _reject;
```

Comment:
It can be reduced to the following code then: 
```js
function inner() {
   if (typeof arguments[arguments.length - 1] === 'function') {
     return fn.apply(this, arguments);
   }

   const args = Array.from(arguments);

   return new Promise((resolve, reject) => {
     args.push(single ? resolve : resolver);
     fn.apply(this, args);

     function resolver(err) {
       if (err) {
         return reject(err);
       }
       switch (arguments.length) {
         case 0: return resolve();
         case 1: return resolve();
         case 2: return resolve(arguments[1]);
       }
       return resolve(Array.from(arguments).slice(1));
     }
   });
 }
```